cirq/vis/state_histogram.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tool to visualize the results of a study."""
+
+from typing import Union, Optional
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+import cirq.study.result as result
+
+
+def get_state_histogram(result: 'result.Result') -> np.ndarray:
+    num_qubits = sum([value.shape[1] for value in result.measurements.values()])
+    states = 2 ** num_qubits
+    values = np.zeros(states)
+    # measurements is a dict of {measurement gate key:
+    #                            array(repetitions, boolean result)}
+    # Convert this to an array of repetitions, each with an array of booleans.
+    # e.g. {q1: array([[True, True]]), q2: array([[False, False]])}
+    #      --> array([[True, False], [True, False]])
+    measurement_by_result = np.hstack(list(result.measurements.values()))
+
+    for meas in measurement_by_result:
+        # Convert each array of booleans to a string representation.
+        # e.g. [True, False] -> [1, 0] -> '10' -> 2
+        state_ind = int(''.join([str(x) for x in [int(x) for x in meas]]), 2)
+        values[state_ind] += 1
+    return values
+
+
+def plot_state_histogram(
+    values: Union['result.Result', np.ndarray], ax: Optional['plt.Axis'] = None
+) -> 'plt.Axis':
+    """Plot the state histogram from a single result with repetitions.
+
+    States is a bitstring representation of all the qubit states in a single
+    result.
```

Comment:
This comment should be updated to reflect the new accepted types for `values`.

Reply:
done